File: 3321-type-of-triangle/type-of-triangle.cpp
class Solution {
public:
    string triangleType(vector<int>& nums) {
        unordered_set<int>s1;
        int count =0;
        int n = nums.size();

        for(int i=0; i<n;i++){
            if(s1.count(nums[i])==1) count++;
            s1.insert(nums[i]);
        }
          if (nums[0] + nums[1] <= nums[2] || nums[0] + nums[2] <= nums[1] || nums[1] + nums[2] <= nums[0]) {
            return "none";
        }
        if(count == 2) return "equilateral";
        else if(count == 1) return "isosceles";
        else if(count == 0) return "scalene";
        else return "none";
        
    }
};

File: 3321-type-of-triangle/type-of-triangle.java
class Solution {
    public String triangleType(int[] nums) {
         if (nums.length != 3) return "none"; // Invalid input

        int a = nums[0], b = nums[1], c = nums[2];

        // Triangle Inequality Theorem
        if (a + b <= c || a + c <= b || b + c <= a) {
            return "none";
        }

        if (a == b && b == c) return "equilateral";
        else if (a == b || b == c || a == c) return "isosceles";
        else return "scalene";
    }
}